backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -775,4 +780,55 @@ IGitMacheteRepositorySnapshot discoverGitMacheteRepository() throws GitMacheteEx
     }
 
   }
+
+  @Getter
+  private static final class GitMacheteNonRootBranchesCreationResult {
+    private final List<GitMacheteNonRootBranch> createdBranches;
+    private final List<String> skippedBranchNames;
+
+    private GitMacheteNonRootBranchesCreationResult() {
+      this.createdBranches = List.empty();
+      this.skippedBranchNames = List.empty();
+    }
+
+    static GitMacheteNonRootBranchesCreationResult empty() {
+      return new GitMacheteNonRootBranchesCreationResult();
+    }
+
+    private GitMacheteNonRootBranchesCreationResult(GitMacheteNonRootBranchesCreationResult prevResult, String skippedBranch) {
```

Comment:
Nit: I think all the 4 existing c'tors can be inlined at their call sites and just replaced with a call to a private `@RequiredArgsConstructor` to reduce verbosity